src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -221,20 +251,16 @@ object DataTypes {
         .withFieldConst(_.aggregation, aggregation.toList.flatten.map(_.toAggregation))
         .transform
 
-      val nonExistingFields = findNonExistingFields(query, entity, tezosPlatformDiscovery)
-      val invalidTypeAggregationField = findInvalidAggregationTypeFields(query, entity, tezosPlatformDiscovery)
-
-      for {
-        invalidNonExistingFields <- nonExistingFields
-        invalidAggregationFieldForTypes <- invalidTypeAggregationField
-        updatedQuery <- replaceTimestampInPredicates(entity, query, tezosPlatformDiscovery)
-      } yield {
-        invalidNonExistingFields ::: invalidAggregationFieldForTypes match {
-          case Nil => Right(updatedQuery)
-          case wrongFields => Left(wrongFields)
-        }
+      (findNonExistingFields(query, entity, metadataService),
+      findInvalidAggregationTypeFields(query, entity, metadataService),
+      findInvalidPredicateFilteringFields(query, entity, metadataService),
+      replaceTimestampInPredicates(entity, query, metadataService)).mapN {
+        case (invalidNonExistingFields, invalidAggregationFieldForTypes, invalidPredicateFilteringFields, updatedQuery) =>
```

Comment:
Triviality: not sure, but you might not need to use a `case` to deconstruct the input, the input should be multliple args, and not a tuple... could be wrong though

Reply:
yes, you're right, changed